src/vs/workbench/contrib/update/browser/update.ts

```diff
@@ -538,20 +535,23 @@ export class SwitchProductQualityContribution extends Disposable implements IWor
 				}
 
 				private async selectSettingsSyncService(dialogService: IDialogService): Promise<UserDataSyncStoreType | undefined> {
-					const res = await dialogService.show(
-						Severity.Info,
-						nls.localize('selectSyncService.message', "Choose the settings sync service to use after changing the version"),
-						[
-							nls.localize('use insiders', "Insiders"),
-							nls.localize('use stable', "Stable (current)"),
-							nls.localize('cancel', "Cancel"),
+					const { result } = await dialogService.prompt<UserDataSyncStoreType>({
```

Comment:
@sandy081 fyi please check, this one is a bit more complex adoption of the new dialog api